### DISCUSSION TITLE
A服务调用B服务，B服务抛异常后A服务数据不回滚。A抛异常B正常回滚，其中xid和branch都正常

### DISCUSSION THREAD

**Discussion by Bourne1314** [apache/incubator-seata#4210] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/23721287/146341910-0ec2eaea-473f-46d3-9d1a-c5a99f91fa56.png)
如图，我用A服务通过feign调用B后，B抛异常，A不回滚。
但A抛异常，B能正常回滚。想问一下大概是啥原因鸭。试了很多次，B都能收到xid，但就是抛异常的时候，A检测不到且不回滚。
系统没有服务熔断。**_### 感谢开发者大大。_**

**Comment by funky-eyes** [comment, marked as answer]
b抛异常可能只是打印了异常，并没有抛给A，你可能存在trycatch或者全局异常捕获器的操作导致的

**Comment by huatuan119** [comment]
我也是,要是能找到看到我就不用折腾这么久了
